tests/test_distgit/test_generic_distgit.py

```diff
@@ -22,194 +24,286 @@ def test_init(self):
         self.assertIsInstance(self.dg, distgit.DistGitRepo)
 
     def test_init_with_branch_override(self):
-        metadata = mock.Mock()
-        metadata.runtime.branch = "original-branch"
+        metadata = flexmock(runtime=flexmock(branch="original-branch"),
+                            config=flexmock(distgit=flexmock(branch=distgit.Missing)),
+                            name="_irrelevant_",
+                            logger="_irrelevant_")
```

Comment:
That's something else I dislike on my implementation ... I see the value of not hiding anything, and it resonates with the Zen of Python:
> Explicit is better than implicit.

but I feel it pollutes the code and hides the properties that are actually relevant for the current test.
What are your thoughts in this matter ?

Reply:
I'm all for clean but understandable. Personally I feel some python programmers go overboard with explicit over implicit... I'd rather it be readable.

Reply:
great, i also share this opinion. i'll try to come up with something cleaner soon.